src/Buoyancy/gravitational_acceleration.jl

```diff
@@ -0,0 +1,7 @@
+@inline g_x(g::Number) = 0
+@inline g_y(g::Number) = 0
+@inline g_z(g::Number) = g
+
+@inline g_x(g::Tuple) = g[1]
+@inline g_y(g::Tuple) = g[2]
+@inline g_z(g::Tuple) = g[3]
```

Comment:
Nice solution!

Reply:
Although I guess this is circumvented if we just convert `g` to a tuple in the constructor, right? This would also simplify the struct since we wouldn't need the parameters `FT` and `G` separately.

Reply:
May need `@inbounds`.

Reply:
Yeah I initially attempted to make `gravitational_acceleration` into a permanent 3-tuple but figured that using `g_x`, `g_y`, `g_z` would allow us to easily dispatch on `::Function` and `::Tuple{Function,Function,Function}` in the future in case we wanted g to vary with x, y, z.